### PR TITLE
fixes error when quitting app during download

### PR DESCRIPTION
**Pull request by patnir**
Before: 

https://github.com/iron-fish/ironfish-node-app/assets/13268167/4aa729ba-f7e9-4a03-96f2-3ca5ce05f2d6

After: 

https://github.com/iron-fish/ironfish-node-app/assets/13268167/2fc6e4a7-6959-4162-a42b-c1ebbe519a0e

### CODE REVIEW

File: main/background.ts [hunk 1]
```diff
@@ -108,3 +111,11 @@ app.on("will-quit", (e) => {
     });
   }
 });
+
+app.on("before-quit", async (e) => {
+  if (ironfish.snapshotManager.started) {
+    e.preventDefault();
```
Comment (jowparks) [inline main/background.ts:117]: What is `preventDefault`?

File: main/api/snapshot/snapshotManager.ts [hunk 2]
```diff
@@ -34,6 +35,19 @@ export class SnapshotManager {
...
 
+  async stop(): Promise<void> {
+    if (!this.snapshotDownloader || !this.started) {
+      return;
+    }
+
+    this.snapshotDownloader.stopDownload();
+    this.started = false;
+
+    this.snapshotPromise.reject(
```
Comment (jowparks) [inline main/api/snapshot/snapshotManager.ts:46]: Why are you rejecting on `stop()`? The point of calling this method is to stop, so you are successfully executing method.

File: main/background.ts (hunk 1 above)
Comment (patnir) [reply main/background.ts:117]: The normal behavior here is to just quit the app. More info here: https://www.electronjs.org/docs/latest/api/app#event-before-quit

So this stops that behavior using preventDefault and stops the download before proceeding the quit the app (app.quit two lines below this one)

Comment (patnir) [reply main/background.ts:117]: docs on preventDefault: https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault

File: main/background.ts [hunk 3]
```diff
@@ -33,6 +33,9 @@ async function createWindow(handler: ReturnType<typeof createIPCHandler>) {
   });
 
   window.once("closed", () => {
+    if (window.isDestroyed()) {
+      return;
+    }
```
Comment (dguenther) [inline main/background.ts:38]: Looking at [the source for electron-trpc](https://github.com/jsonnull/electron-trpc/blob/784bbb0cf45b0304c9f55bd9827a51c8dd3a5b6a/packages/electron-trpc/src/main/createIPCHandler.ts#L50), it looks like detachWindow is already called when windows are closed, so I think it's okay to remove our `window.once("closed"` entirely. 

From testing on macOS by clicking the close button, it seems like `window.isDestroyed` returns true at this point, so it ends up not doing anything.

**Review by dguenther — CHANGES_REQUESTED** [review]

**Pull request closed without merging**